Firestore/core/src/firebase/firestore/remote/stream.h

```diff
@@ -37,6 +37,14 @@ namespace firebase {
 namespace firestore {
 namespace remote {
 
+class StreamListener {
```

Comment:
@wilhuff Gil, I tried out an approach to listener closer to Android/Web clients, like we discussed in the `Stream` PR. See the second commit in the branch for a diff.
 
I personally prefer the previous version to the new one:
* there is now a parallel class hierarchy of listeners;
* the responsibility for notifying listeners is now spread across base `Stream` and derived `WatchStream`. Worse, there appears to be no conceptual reason for the way the notifications are split across classes;
* `Stream` now has an additional responsibility even though it already does quite a lot;
* inheriting/creating listener classes is more boilerplate.

So these are my thoughts, but if you find that consistency with other platforms trumps these considerations, I won't argue.